pages/games/[id]/index.js

```diff
@@ -0,0 +1,100 @@
+import Link from "next/link";
+import { useRouter } from "next/router";
+import useSWR from "swr";
+import styled from "styled-components";
+
+export default function DetailsPage() {
+  const router = useRouter();
+  const { id } = router.query;
```

Comment:
your "id" will be undefined in an edge case, when the details page is reloaded or opened directly in the browser. This is the case because then the page is pre-rendered on the server and in that case all router.query parameters are unknown. Only after a rerender which is done on the client the correct id is then added. 

Please add a check if the id is undefined and if so return a loading state.